.github/workflows/ci.yml

```diff
@@ -78,3 +78,19 @@ jobs:
 
       - name: "Coverage"
         run: echo "$(cat ./.phpunit/coverage.txt)"
+
+      - name: Make code coverage badge
+        uses: timkrase/phpunit-coverage-badge@v1.2.1
+        with:
+          coverage_badge_path: .github/coverage.svg
+          # push badge later on
+          push_badge: false
+
+      - name: Git push badges to image-data branch
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          publish_dir: .github
+          publish_branch: image-data
+          github_token: ${{ secrets.PHP_GA4_TOKEN }}
```

Comment:
If I read it correctly then github_token is for Pages, and I have not added pages for this branch. It seems, in the doc, that you should be able to use as `personal_token`.

Try `personal_token: ${{ secrets.PHP_GA4_TOKEN }}`
~ref https://github.com/peaceiris/actions-gh-pages#supported-tokens

Otherwise, would you mind explaining what steps you took to make this work?

Reply:
I've setup as follows, go to repository `Settings` > `Actions` > `General` > `Workflow permissions` > `Read and write permissions`

Reply:
I've updated the token as you suggested. If it doesn't work, we should use the original token as I know for sure that it works.

Reply:
I checked the path you provided and it is on `Read and write permissions`, have been all the time. This is why I'm confused, the only thing blocking is that I have to "approve" all merge requests to run the tests as I would like to add "allow actions to merge pull requests"

Reply:
The full setup is

# Actions permissions

`Allow aawnu, and select non-aawnu, actions and reusable workflows`
-> `Allow actions created by GitHub`
```
easimon/wipe-cache/*,
keinos/gh-action-hash-for-cache/*,
peaceiris/actions-gh-pages/*,
shivammathur/setup-php/*,
timkrase/phpunit-coverage-badge/*,
```

`Require approval for all outside collaborators`

`Allow GitHub Actions to create and approve pull requests`

Reply:
Here are my settings, I didn't touch anything else from the default repo settings.

![2023-03-23_181840](https://user-images.githubusercontent.com/13252042/227232524-7b3a0d26-e322-4e96-b2fb-71c1d6a67190.png)

Reply:
Try and change it back to the github token, and then I'll see if I can make the settings work properly and rerun the action.